### PR TITLE
Export the hostname as environment variable

### LINKED ISSUE #8470: Set default node name to hostname
I can't find any discussion on it. Can this be done? It would be convenient.

### PR DESCRIPTION
**Pull request by AndreKR**
and provide an example on how to use it in the config

Closes #8470

### CODE REVIEW

**Comment by clintongormley** [comment]
Hi @AndreKR 

Thanks for the PR.  The only bit I'm not sure about is adding it to the config file, because using the hostname there can be problematic if you run more than one ES instance on the same node.

Perhaps it should be added to the reference docs instead, with the appropriate warning: http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/setup-configuration.html#node-name

**Comment by AndreKR** [comment]
Done.

File: bin/elasticsearch.bat [hunk 1]
```diff
@@ -33,6 +33,9 @@ FOR /F "usebackq tokens=1* delims= " %%A IN (!params!) DO (
 	)
 )
 
+set HOSTNAME=
+for /f "delims=" %%a in ('hostname') do @set HOSTNAME=%%a
```
Comment (tlrx) [inline bin/elasticsearch.bat]: Can you please set instructions in uppercase to stick with the rest of the file?

Also, I'm not sure this works on older Windows like XP. Do you think you can get it work on it? That would be nice.

**Comment by tlrx** [comment]
@AndreKR Thanks for your contribution.

I made some comments. Can you also please squash all the commits together? Thanks.

File: bin/elasticsearch.bat (hunk 1 above)
Comment (AndreKR) [inline bin/elasticsearch.bat]: Argh, didn't see the uppercase, fixed.
Well, I wasn't successful running elasticsearch on Windows 2000 ("Unsupported major.minor version 51.0") but I can confirm that the solution for setting the hostname works:

**Comment by AndreKR** [comment]
Done.

**Comment by tlrx** [comment]
I just tested on Windows XP and unfortunately it does not work and prevents Elasticsearch from starting.

I think it can be fixed like this:
```
SET ES_HOSTNAME=''
FOR /F "delims=" %%A IN ('hostname') DO (
    SET ES_HOSTNAME=%%A
)
```
What do you think? Does it work on your platforms?

**Comment by AndreKR** [comment]
Ok, I looked into the issue and indeed on Windows XP it didn't work, even though on Windows 2000 and Windows 7 it did. The actual problem were the missing quotes, with them it works on all three platforms. I pushed the change (and rebased the branch).

**Comment by clintongormley** [comment]
@tlrx please could you take a look

**Comment by tlrx** [comment]
@AndreKR it takes some time but be sure that I'd like to merge this pull request :) Have you considered using %COMPUTERNAME% variable instead of hostname? Do you think it's better?

**Comment by AndreKR** [comment]
By using %COMPUTERNAME%, are you trying to solve a specific problem or just considering it as a cleaner approach? Honestly, I hadn't considered that, because I implemented the Linux variant first and then just looked into a parallel approach for Windows.
It would certainly have the advantage of making the batch file easier to read, so I'm going to implement that change.
Do you mean on Windows people should just use COMPUTERNAME or do we want the batch file to copy it over into HOSTNAME so that the elasticsearch config file and documentation is portable across platforms?

**Comment by tlrx** [comment]
> By using %COMPUTERNAME%, are you trying to solve a specific problem or just considering it as a cleaner approach?

Just a cleaner approach since `%COMPUTERNAME%` seems to be consistent accross windows versions.

> Do you mean on Windows people should just use COMPUTERNAME or do we want the batch file to copy it over into HOSTNAME so that the elasticsearch config file and documentation is portable across platforms?

Copying it into `HOSTNAME` sounds good.

Thanks for making this change! I'll test it again and merge the pr.

**Comment by AndreKR** [comment]
Changed.

**Comment by tlrx** [comment]
@AndreKR Finally merged, thanks! I changed the wording a bit in b3d91b1cbb94db950f6007df49beff5f5d167b07

**Comment by agirbal** [comment]
awesome thanks for fixing!

**Pull request merged**